src/IO/S3/Client.cpp

```diff
@@ -267,17 +267,32 @@ void Client::setKMSHeaders(RequestType & request) const
     }
 }
 
+
+template <typename RequestType>
```

Comment:
The function has to be put into an anonymous namespace.

Reply:
done